packages/aws-cdk-lib/aws-ecs/lib/cluster.ts

```diff
@@ -66,11 +66,12 @@ export interface ClusterProps {
   readonly enableFargateCapacityProviders?: boolean;
 
   /**
-   * If true CloudWatch Container Insights will be enabled for the cluster
+   * If enabled CloudWatch Container Insights will be enabled for the cluster.
+   * If enhanced, enhanced CloudWatch Container Insights will be enabled.
    *
    * @default - Container Insights will be disabled for this cluster.
    */
-  readonly containerInsights?: boolean;
+  readonly containerInsights?: ContainerInsights;
```

Comment:
Hello unfortunately, this would be a breaking change so we can't accept this PR. My suggestion would be marking this property as deprecated and create a new `containerInsightsV2` property that takes the ENUM or ENUM-like class (recommended) instead.